tornado/httputil.py

```diff
@@ -230,13 +235,148 @@ def parse(cls, headers):
            mix of `KeyError`, and `ValueError`.
 
         """
-        h = cls()
+        h = HTTPHeaders()
         for line in headers.splitlines():
             if line:
                 h.parse_line(line)
         return h
 
-    # MutableMapping abstract method implementations.
+    def pop(self, key, default=__marker):
+        '''D.pop(k[,d]) -> v, remove specified key and return the corresponding value.
+          If key is not found, d is returned if given, otherwise KeyError is raised.
+        '''
+        try:
+            value = self._dict[_normalized_headers[key]]
```

Comment:
Shouldn't this pop the value from `_as_list`?

Reply:
See line 255